api/v1alpha1/project_webhook.go

```diff
@@ -62,7 +63,11 @@ func (r *Project) ValidateCreate() error {
 func (r *Project) ValidateUpdate(old runtime.Object) error {
 	projectlog.Info("validate update", "name", r.Name)
 
-	_, err := r.IsValidUpdate()
+	oldProject, ok := old.(*Project)
+	if !ok {
+		log.Error(ok, "Error in finding the requested project")
```

Comment:
This should return an error, don't just log it. Something like this should do
`return fmt.Errorf("Error casting old runtime object to %T from %T", oldProject, old)`